### PR TITLE
[MM-18819] Backport desktop notification changes to 5.14

### PR DESCRIPTION
**Pull request by deanwhillier**
#### Summary
Backport of functionality required by Desktop v4.3 to support Desktop Notifications. See ticket and PR below

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18819
https://github.com/mattermost/mattermost-webapp/pull/3718

### CODE REVIEW

File: components/logged_in/logged_in.jsx [hunk 1]
```diff
@@ -71,10 +74,16 @@ export default class LoggedIn extends React.PureComponent {
         window.addEventListener('focus', this.onFocusListener);
         window.addEventListener('blur', this.onBlurListener);
 
-        // Listen for user activity updates from external sources via the webapp connector
-        if (webappConnector.active) {
```
Comment (jespino) [inline components/logged_in/logged_in.jsx]: Is this change backward compatible, do we need backward compatibility here?

Comment (jespino) [reply components/logged_in/logged_in.jsx]: I'm thinking here about using old desktop versions with new mattermost server, or using old mattermost servers with the new changes in the desktop app.

Comment (deanwhillier) [reply components/logged_in/logged_in.jsx]: Good question and good eye @jespino! That code was the original implementation of a communication bridge between the Webapp and the yet-to-be-released v4.3 version of the Destop app. A security related update to the Desktop app invalidated that approach and was replaced by the update in this PR. So it's never been used and therefore won't have an impact on any version of the Desktop app. 🙂

**Comment by deanwhillier** [comment]
@amyblais, this PR is for backport to v5.14

**Comment by lindy65** [comment]
Thanks @deanwhillier 

Tested on the spinwick on desktop app 4.3 RC2 for desktop notifications with the following settings:
- For all activity
- Only for mentions and direct messages
- Never

and did not find any issues. Notifications work as expected per setting.

**Pull request merged**